src/ui/composables/alignmentContext.js

```diff
@@ -36,32 +36,20 @@ const alignmentMap = new Map();
 export function useAlignment(targetObject, path, openmct) {
   const getAlignmentKeyForPath = () => {
     const keys = Array.from(alignmentMap.keys());
-    if (!keys.length) {
-      return;
-    }
-    return (
-      path
-        //get just the identifiers
-        .map((domainObject) => {
-          return openmct.objects.makeKeyString(domainObject.identifier);
-        })
-        .reverse()
-        // find a match
-        .find((keyString) => {
-          return keys.includes(keyString);
-        })
-    );
+    return path
+      .map((domainObject) => openmct.objects.makeKeyString(domainObject.identifier))
+      .reverse()
+      .find((keyString) => keys.includes(keyString));
   };
 
   // Use the furthest ancestor's alignment if it exists, otherwise, use your own
-  let alignmentKey = getAlignmentKeyForPath(path);
+  let alignmentKey =
+    getAlignmentKeyForPath() || openmct.objects.makeKeyString(targetObject.identifier);
 
-  if (!alignmentKey) {
-    const targetObjectKey = openmct.objects.makeKeyString(targetObject.identifier);
-    alignmentKey = targetObjectKey;
+  if (!alignmentMap.has(alignmentKey)) {
     alignmentMap.set(
-      targetObjectKey,
-      ref({
+      alignmentKey,
+      reactive({
```

Comment:
@ozyx Cool. From what I had read before, reactive() objects cannot be reassigned. Changing their property values should be ok, right?

Reply:
That is correct. Reactive objects cannot be reassigned, but changing their property values is okay. We don't reassign it anywhere in the composable, and when we return the `alignment` object we feed it through the `toRefs()` function which converts reactive objects into objects where each property is a `ref()`. That allows us to destructure the `alignment` object within the setup() function without losing reactivity.